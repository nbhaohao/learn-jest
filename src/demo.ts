export const generateConfig = () => {
  return {
    server: "http://localhost",
    port: 80,
    time: new Date()
  };
};
