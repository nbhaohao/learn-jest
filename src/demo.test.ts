import { generateConfig } from "./demo";

test("测试 generateConfig 函数", () => {
  expect(generateConfig()).toMatchSnapshot({
    time: expect.any(Date)
  });
});

test("测试 generateConfig 函数 inline snapshot", () => {
  expect(generateConfig()).toMatchInlineSnapshot(
    {
      time: expect.any(Date)
    },
    `
    Object {
      "port": 80,
      "server": "http://localhost",
      "time": Any<Date>,
    }
  `
  );
});
