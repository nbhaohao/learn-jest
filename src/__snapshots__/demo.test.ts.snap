// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 generateConfig 函数 1`] = `
Object {
  "port": 80,
  "server": "http://localhost",
  "time": Any<Date>,
}
`;

exports[`测试 generateConfig 函数 2`] = `
Object {
  "port": 80,
  "server": "http://localhost",
  "time": Any<Date>,
}
`;

exports[`测试 generateConfig 函数 inline snapshot 1`] = `
Object {
  "port": 80,
  "server": "http://localhost",
  "time": Any<Date>,
}
`;

exports[`测试 generateConfig 函数1 1`] = `
Object {
  "port": 80,
  "server": "http://localhost",
  "time": Any<Date>,
}
`;
